Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.tsx

```diff
@@ -16,61 +18,198 @@ import styles from './ContestCategories.module.scss';
+
+    const categoriesFlat = useMemo(
+        () => flattenWith(categories, (c) => c.children || null),
+        [ categories ],
+    );
+
+    const getCategoryId = useCallback(
+        (searchedId?: string, searchedValue?: string) => {
+            const filterToFind = possibleFilters
+                .find(({ id, value }) => id.toString() === searchedId || value === searchedValue?.toString()) as IFilter;
+
+            if (isNil(filterToFind)) {
+                return '';
+            }
+
+            const { value } = filterToFind;
+
+            return value;
+        },
+        [ possibleFilters ],
+    );
+
+    const getCurrentNode = useCallback(
+        (searchId?: string, searchValue?: string) => {
+            const categoryId = getCategoryId(searchId, searchValue);
+
+            return categoriesFlat.find(({ id }) => id.toString() === categoryId);
+        },
+        [ categoriesFlat, getCategoryId ],
+    );
 
     const getParents = useCallback(
-        (result: string[], allItems: ITreeItemType[], searchId?: string) => {
+        (result: string[], allItems: ITreeItemType[], searchId?: string, searchValue?: string) => {
             if (isNil(searchId)) {
                 return result;
             }
 
-            const node = allItems.find(({ id }) => id.toString() === searchId);
+            const currentNode = getCurrentNode(searchId, searchValue);
 
-            if (isNil(node)) {
+            if (isNil(currentNode)) {
                 return result;
             }
 
-            if (node.id.toString() === searchId) {
-                result.push(searchId);
-            }
+            result.push(currentNode.id.toString());
 
-            getParents(result, allItems, node.parentId?.toString());
+            getParents(result, allItems, '', currentNode.parentId?.toString());
 
             return result;
         },
-        [],
-    );
-
-    const categoriesFlat = useMemo(
-        () => flattenWith(categories, (c) => c.children || null),
-        [ categories ],
+        [ getCurrentNode ],
     );
 
     const defaultExpanded = useMemo(
         () => getParents([], categoriesFlat, defaultSelected),
         [ defaultSelected, categoriesFlat, getParents ],
     );
 
-    const handleTreeLabelClick = useCallback((node: ITreeItemType) => {
-        const filter = possibleFilters.find(({ value }) => value.toString() === node.id.toString());
-        const category = categoriesFlat.find(({ id }) => id.toString() === node.id.toString());
+    const strategyFilterGroup = useMemo(
+        () => possibleFilters.filter(({ type }) => type === FilterType.Strategy),
+        [ possibleFilters ],
+    );
+
+    const getStrategyFiltersToAdd = useCallback(
+        ({ allowedStrategyTypes }: ICategoryStrategiesTypes) => allowedStrategyTypes
+            ?.map((value) => value.id.toString())
+            .map((x) => strategyFilterGroup.find((y) => y.value === x))
+            .filter((x) => !isNil(x)) as IFilter[],
```

Comment:
This is hard to read, please simplify, e.g. you can filter out null items in the map

Reply:
Left for further discussions.